src/actions/addPackageJsonFunding.ts

```diff
@@ -0,0 +1,124 @@
+import mls from 'multilines'
+import Octokit, { ReposGetResponse } from '@octokit/rest'
+import detectIndent from 'detect-indent'
+
+import { base64 } from '../utils'
+import { getCollectiveWithGithubHandle } from '../collective'
+import { resetBranch } from '../github'
+
+const defaultFundingPackageAsString = `https://opencollective.com/<YOUR-COLLECTIVE-SLUG>`
+
+const BRANCH_NAME = `opencollective-bot/funding_package`
+
+const FILE_PATH = 'package.json'
+
+const PR_TITLE = 'Add FUNDING to package.json'
+
+const PR_BODY = mls`
+  | Add FUNDING to package.json
+  |
+  | Add FUNDING to package.json
+  |
+  | Add FUNDING to package.json
+  `
+
+const FUNDING_DEFAULT_FILE_CONTENT = defaultFundingPackageAsString
+
+function setIndent(packageJSON: string): string {
+  return detectIndent(packageJSON).indent || '  '
+}
+
+function updatePackageJson(packageJSON: string, fundingUrl: string) {
+  const indent = setIndent(packageJSON)
+  const obj = JSON.parse(packageJSON)
+  obj['funding'] = {
+    type: 'opencollective',
+    url: fundingUrl,
+  }
+  return JSON.stringify(obj, null, indent)
+}
+
+export { setIndent }
+export default async function addPackageJsonFunding(
+  github: Octokit,
+  repoResponse: ReposGetResponse,
+) {
+  const owner = repoResponse.owner.login
+  const repo = repoResponse.name
+
+  // Check if package.json file is existing
+  const packageJsonFile = await github.repos
+    .getContents({
+      owner,
+      repo,
+      path: 'package.json',
+    })
+    .then((res: any) => res.data)
+    .catch(() => null)
+
+  if (!packageJsonFile) {
+    console.log(`There is not package.json on ${owner}/${repo}`)
+    return
+  }
+
+  // Content Base64 Decoding
+  const packageJsonContent = Buffer.from(
+    packageJsonFile.content,
+    'base64',
+  ).toString()
+
+  // Check if package.json already contains funding property
+  if (packageJsonContent.includes('"funding":')) {
```

Comment:
While it works, let's do that by detecting the property once the JSON is parsed, no?